pkg/serializer/func_queue.go

```diff
@@ -81,6 +85,18 @@ func (fq *FunctionQueue) Stop() {
 	close(fq.stopCh)
 }
 
+// Wait until the FunctionQueue is stopped, or the specified context deadline
+// expires. Returns the error from the context, or nil if the FunctionQueue
+// was completed.
+func (fq *FunctionQueue) Wait(ctx context.Context) error {
+	select {
+	case <-fq.stopCh:
```

Comment:
fun fact of the day, if both fq.stopCh and ctx.Done are closed at the exact same time, golang will randomly pick one of these 2 cases, so to be precise you might want to return `ctx.Err()` on line 97 to ensure that in case both channels got closed at the same time and the fq.stopCh case is "picked" that we still return the error for the context

Reply:
o_o;;;  OK yeah that would be more deterministic. I'll fix it up.